### ISSUE TITLE
InvalidOperationException: StatusCode cannot be set because the response has already started in middleware after hotchocollate upgrade

### ISSUE THREAD

**Issue by martinsura** [ChilliCream/hotchocolate#1667]
**Describe the bug**
Hello,

I have simple validation middelware:
```
    public class GraphQLValidationMiddleware
    {
        private readonly FieldDelegate _next;

        public GraphQLValidationMiddleware(FieldDelegate next)
        {
            _next = next;
        }

        public async Task InvokeAsync(IMiddlewareContext context)
        {
            var input = context.Argument<object>("input");
            if (input != null)
            {
                var validationContext = new ValidationContext(input, serviceProvider: null, items: null);
                var validationResults = new List<ValidationResult>();
                Validator.TryValidateObject(input, validationContext, validationResults, true);
                if (validationResults.Count > 0)
                {
                    var error = ErrorBuilder.New().SetCode("400").SetMessage("Validation Error").SetExtension("modelErrors", validationResults).Build();
                    context.Result = error;
                    return;
                }
            }
            await _next(context);
        }
    }

```

After upgrading HotChocolate to version **1.0.0-preview.120** got 
`System.InvalidOperationException: StatusCode cannot be set because the response has already started.` exception

Got same error even if i set context.ReportError();.

Its asp net core 3.1 applications (Windows 10)

```
  <PropertyGroup>
    <TargetFramework>netcoreapp3.1</TargetFramework>
  </PropertyGroup>
    <PackageReference Include="HotChocolate.AspNetCore" Version="11.0.0-preview.120" />
    <PackageReference Include="HotChocolate.AspNetCore.Authorization" Version="11.0.0-preview.120" />
    <PackageReference Include="HotChocolate.AspNetCore.Playground" Version="11.0.0-preview.120" />
  <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="3.1.3" />
```

**11.0.0-preview.70** works fine.

**Comment by martinsura** [comment]
Aaand now i got 
```
System.NotSupportedException: 
The specified type System.ComponentModel.DataAnnotations.ValidationResult is not supported by the result serializer.
```
**11.0.0-preview.123**

**Comment by michaelstaib** [comment]
This one is now fixed with 127 or 10.4.2
